projects/gloo/cli/pkg/cmd/gatewayapi/convert/command.go

```diff
@@ -0,0 +1,223 @@
+package convert
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/solo-io/gloo/pkg/utils/envoyutils/admincli"
+	"github.com/solo-io/gloo/pkg/utils/kubeutils/kubectl"
+	"github.com/solo-io/gloo/pkg/utils/kubeutils/portforward"
+	"github.com/solo-io/gloo/pkg/utils/requestutils/curl"
+	"github.com/solo-io/gloo/projects/gloo/cli/pkg/cmd/options"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	RandomSuffix = 4
+	RandomSeed   = 1
+)
+
+func RootCmd(op *options.Options) *cobra.Command {
+	opts := &Options{
+		Options: op,
+	}
+	cmd := &cobra.Command{
+		Use:   "convert",
+		Short: "Convert Gloo Edge APIs to Gateway API",
+		Long:  "Convert Gloo Edge APIs to Gateway APIs by either providing Kubernetes YAML files or a Gloo Gateway input snapshot.",
+		Example: `# This command converts Gloo Edge APIs to Kubernetes Gateway API YAML files and places them in the '--output-dir' directory, grouped by namespace.
+# Example command to generate Gateway API YAML files from a Gloo Gateway snapshot that is retrieved from a running 'gloo' pod. The 'output-dir' must not exist.
+  glooctl gateway-api convert --gloo-control-plane deploy/gloo --output-dir ./_output
+
+# Generate Gateway API YAML files from a single Kubernetes YAML file. The 'output-dir' must not exist.
+  glooctl gateway-api convert --input-file gloo-yamls.yaml --output-dir ./_output
+
+# To delete and recreate the content in the `output-dir`, add the `delete-output-dir` option.
+  glooctl gateway-api convert --input-file gloo-yamls.yaml --output-dir ./_output --delete-output-dir
+
+# To generate gateway api by a single kubernetes yaml file but place all the output configurations in the same file.
+  glooctl gateway-api convert --input-file gloo-yamls.yaml --output-dir ./_output --retain-input-folder-structure
+
+# To load a bunch of *.yaml or *.yml files in nested directories. You can also retain the same file structure which is helpful in CI/CD pipelines.
+  glooctl gateway-api convert --input-dir ./gloo-configs --output-dir ./_output --retain-input-folder-structure
+
+# Configuration can also be pulled an translated directly from a running Gloo Gateway Instance (version 1.17+).
```

Comment:
To download a Gloo Gateway snapshot from a running 'gloo' pod (verison 1.17+) and generate Gateway API YAML files from that snapshot. 


glooctl gateway-api convert --gloo-control-plane deploy/gloo --output-dir ./_output